pkg/openstack/nova.go

```diff
@@ -58,6 +58,10 @@ func ReconcileNova(ctx context.Context, instance *corev1beta1.OpenStackControlPl
 		return ctrl.Result{}, nil
 	}
 
+	if instance.Spec.Nova.Template == nil {
+		instance.Spec.Nova.Template = &novav1.NovaSpec{}
```

Comment:
nit: NovaSpec is overkill, only the NovaSpecCore part of it will be used after the assignment so you can simply create NovaSpecCore right away.

Reply:
right, but then we'd have to change https://github.com/openstack-k8s-operators/openstack-operator/blob/main/apis/core/v1beta1/openstackcontrolplane_types.go#L537 . I did not want to do it as part of this PR. It also requires a change on how the template gets assigned to the service spec in 
 https://github.com/openstack-k8s-operators/openstack-operator/blob/main/pkg/openstack/nova.go#L321 . We'd then should add a func which converts the NovaSpecCore to a NovaSpec

Reply:
Hm. It feels like a bug from the PR that added the usage of the Core Specs. I agree to do a fix separately.